### PR TITLE
Handle all dictionary exception more properly

### PR DESCRIPTION
**Pull request by foryou2030**
# Why rasied this pr?

When using all dictionary, if we give a wrong dictionary file path, or dictionary file include bad record,
carbon should deal with them properly.
# How to solve?

1.when give a wrong dictionary file path, throw file not found exception
2.when dictionary file include bad record,log error, and replace the bad record as default value.

### CODE REVIEW

**Comment by foryou2030** [comment]
Passed CI test, pls check: http://136.243.101.176:8080/job/ApacheCarbonManualPRBuilder/24/

File: integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala [hunk 1]
```diff
@@ -629,22 +631,32 @@ object GlobalDictionaryUtil extends Logging {
...
+          logWarning("[ALL_DICTIONARY] No dictionary files found or empty dictionary files! " +
+            "Won't generate new dictionary.")
+          false
+        }
       } else {
-        logInfo("No dictionary files found or empty dictionary files! " +
-          "Won't generate new dictionary.")
-        false
+        throw new FileNotFoundException(
+          "[ALL_DICTIONARY] The given dictionary file path not found!")
```
Comment (Vimal-Das) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala]: Correct english grammar in the log messages.
The given dictionary file path not found! =>The given dictionary file path **is** not found!

Comment (foryou2030) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala]: ok, fixed

File: integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala [hunk 2]
```diff
@@ -591,29 +591,31 @@ object GlobalDictionaryUtil extends Logging {
...
       // group by column index, and filter required columns
       val requireColumnsList = requireColumns.toList
       allDictionaryRdd = basicRdd
         .groupByKey()
-        .map(x => (csvFileColumns(x._1), x._2))
         .filter(x => requireColumnsList.contains(x._1))
     } catch {
       case ex: Exception =>
-        logError("read local dictionary files failed")
+        logError("[ALL_DICTIONARY] Read dictionary files failed. Caused by" + ex.getMessage)
```
Comment (sujith71955) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala]: ALL_DICTIONARY this term is really required in this context?

File: integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala [hunk 3]
```diff
@@ -629,22 +631,32 @@ object GlobalDictionaryUtil extends Logging {
     // filepath regex, look like "/path/*.dictionary"
     if (filePath.getName.startsWith("*")) {
       val dictExt = filePath.getName.substring(1)
-      val listFiles = filePath.getParentFile.listFiles()
-      if (listFiles.exists(file =>
-        file.getName.endsWith(dictExt) && file.getSize > 0)) {
-        true
+      if (filePath.getParentFile.exists()) {
+        val listFiles = filePath.getParentFile.listFiles()
```
Comment (sujith71955) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala]: what if filePath.getParentFile returns null?

Comment (foryou2030) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala]: handled

File: integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala (hunk 2 above)
Comment (foryou2030) [inline integration/spark/src/main/scala/org/apache/carbondata/spark/util/GlobalDictionaryUtil.scala]: ok, it no required, i have removed

**Comment by sujith71955** [comment]
LGTM

**Pull request closed without merging**